packages/slice-machine/src/domain/group.ts

```diff
@@ -0,0 +1,89 @@
+import { NestableWidget } from "@prismicio/types-internal/lib/customtypes";
+
+import { GroupSM } from "@lib/models/common/Group";
```

Comment:
This is tricky to do now because `domain/slice.ts` always works with `ComponentUI` (essential `SliceSM` by a different name). We would need to do `Groups.toSM` and `Groups.fromSM` in multiple areas.

I think we don't see type errors now because the slice-related code uses `any`. I suspect we have some incorrect code though.

I agree we should remove all `*SM` types as it's a big source of indirection. Would you be okay with me refactoring `domain/slice.ts`, and the related `domain/group.ts`, in a separate PR?

Reply:
I edited `domain/group.ts` to use `Group` instead of `GroupSM`. You can see the commit here: https://github.com/prismicio/slice-machine/pull/1335/commits/d377fe21578eb3fc0a25c60f3beadac36d4f8f4a

I needed to add `Groups.toSM` and `Groups.fromSM` when data goes in and out of the helpers. Although it seems a bit messy now, it should removing `*SM` types easier in the future.